### DISCUSSION TITLE
How to Access request.session.get("xxx") in Marimo Notebook ?

### DISCUSSION THREAD

**Discussion by SimonYang-CS** [marimo-team/marimo#3280] [Q&A] [answered]
Hi Marimo Team,  

First of all, thank you for creating such a great product! Marimo has been incredibly useful tool.  

We have successfully protected our Marimo notebook using a custom AuthMiddleware in our FastAPI application. 

However, we are facing an issue with accessing the session data within the Marimo notebook. 
Specifically, we need to retrieve the email stored in the session using request.session.get("xxx").  

Here is a brief overview of our setup:  
We have implemented a custom AuthMiddleware to handle authentication.
The middleware stores the user info the session.
We need to access this email within the Marimo notebook cells.

Could you please provide guidance on how to access request.session.get("xx") within the Marimo notebook?  

Thank you for your help!

**Comment by mscolnick** [comment]
Hi @SimonYang-CS, it is not currently possible. But it is something we can look into adding under `mo.app_meta()`. 

1) Would you be open to making a contribution for this?
2) If not, could you file a feature-request under the issues tab?

**Reply by SimonYang-CS** [reply, marked as answer]
This issue has been fixed in 0.11.6.

1. Add customer AuthMiddleware

`
class AuthMiddleware(BaseHTTPMiddleware):
    async def dispatch(self, request: Request, call_next):
        request.scope["user"] = {
            "is_authenticated": True,
            "username": "email",
        }

        request.scope["meta"] = {
            "email": "god",
            "groups": ["admin", "user"],
        }

        print("in AuthMiddleware")
        return await call_next(request)
`

2. Access data in notebook

`
@app.cell
def cell_user_group(mo):
    meta = mo.app_meta().request.meta
    email = meta.get("email")
    groups = meta.get("groups")
    meta_md = mo.md(f"meta email: {email} groups: {groups}")
    meta_md

    return email, groups
`
